url2a908001cd_malictus_klang_tgz-pJ8-malictus_klang_test_PrimitiveTestJ8/src/malictus/klang/chunk/ZTXTChunk.java

```
@@ -0,0 +1,158 @@
+	 * @throws IOException if any read/write error occurs
+	 */
+	public void reparseChunkPrimitives(RandomAccessFile raff) throws IOException {
+		if ((this.primitives == null) || (this.primitives.size() == 0)) {
+			//nothing to do!
+			return;
+		}
+		Primitive p1 = primitives.get(0).getPrimitive();
+		Primitive p2 = primitives.get(1).getPrimitive();
+		Primitive p3 = primitives.get(2).getPrimitive();
+		long oldsize = this.getDataEndPosition() - this.getDataStartPosition();
+		long pos = 0;
+		try {
+			long newsize = 0;
+			if (p1.valueExists()) {
+				newsize = p1.writeValueToString().getBytes(((StringPrim)p1).getTextEncoding()).length + 1;
+			}
+			if (p2.valueExists()) {
+				newsize = newsize + 1;
+			}
+			if (p3.valueExists()) {
+				newsize = newsize + p3.writeValueToString().getBytes(((StringPrim)p3).getTextEncoding()).length;
+			}
+			if (this.getParentChunk() != null) {
+				if (this.getParentChunk().usesPadByte()) {
+					newsize = KlangUtil.adjustForPadByte(newsize);
+				}
+			}
+			long totnewsize = newsize + (this.getEndPosition() - this.getStartPosition()) - (this.getDataEndPosition() - this.getDataStartPosition());
+			if (newsize < oldsize) {
+				this.chunkIsAboutToChangeSize(totnewsize, raff);
+				KlangUtil.deleteFromFile(this.getParentFile().getFile(), this.getDataStartPosition(), this.getDataStartPosition() + (oldsize - newsize));
+			} else if (newsize > oldsize) {
+				this.chunkIsAboutToChangeSize(totnewsize, raff);
+				int amt = (int)(newsize - oldsize);
+				byte[] newbyte = new byte[amt];
+				KlangUtil.insertIntoFile(newbyte, this.getParentFile().getFile(), this.getDataStartPosition());
+			}
+			raff.seek(this.getDataStartPosition());
+			pos = raff.getFilePointer();
+			//write values back to the file
+			p1.writeValueToFile(raff);
+			if (p2.valueExists()) {
+				p2.writeValueToFile(raff);
+			}	
+			if (p3.valueExists()) {
+				p3.writeValueToFile(raff);
+			}
+		} catch (BadValueException err) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 148.